### DISCUSSION TITLE
[ERROR] Module build failed: Module not found: @symfony/stimulus-bundle

### DISCUSSION THREAD

**Discussion by BAD-WOLF** [symfony/symfony#53463] [Q&A] [answered]
I'm creating a project using Symfony and I'm having problems with the `@symfony/stimulus-bundle` package reference.

Apparently this package is not available on npm and yarn, I installed it via composer as the documentation recommends.

These are the steps I reproduced until I came across the error:

#### Creating Symfony App

```bash
  composer create-project symfony/skeleton symfony_skeleton
```

### Package installation

```bash
  composer require webapp
```

```bash
  composer require symfony/asset-mapper symfony/asset symfony/twig-pack symfony/stimulus-bundle
```

```bash
  composer require webpack
```

### The problem

When I run `yarn watch` i'm facing this error:

###### I ran the commands before running `yarn watch`: (`bin/console asset-map:compile && yarn install`)

```bash
yarn run v1.22.21
$ encore dev --watch
Running webpack ...

 ERROR  Failed to compile with 1 errors                                                                                        20:21:03

Module build failed: Module not found:
"./assets/bootstrap.js" contains a reference to the file "@symfony/stimulus-bundle".
This file can not be found, please check it for typos or update it if the file got moved.

Entrypoint app [big] 671 KiB = runtime.js 14.6 KiB vendors-node_modules_hotwired_stimulus_dist_stimulus_js-node_modules_core-js_modules_es_date_-bf8a9b.js 634 KiB app.css 735 bytes app.js 21.6 KiB
webpack compiled with 1 error

```
![Captura de tela de 2024-01-08 20-21-17](https://github.com/symfony/symfony/assets/88906998/b6bd573a-39bd-434f-ba2c-c7fc97522c69)

Bootstrap file which was mentioned in the log above

```js
import { startStimulusApp } from '@symfony/stimulus-bundle';

const app = startStimulusApp();
// register any custom, 3rd party controllers here
// app.register('some_controller_name', SomeImportedController);
```

I found someone on Stack Overflow with a slightly different context but dealing with the same error and he reports that the problem was resolved by downgrading from version `7.0` to `6.4`

> "The problem encountered was that I was using Symfony 7.0.1, while this method works for Symfony 6.4"

Is there any way to get around this during the project creation and dependency installation process? I wanted to keep my project in version `7.0` and find a way to solve the problem

### References

[Installing & Setting up the Symfony Framework](https://symfony.com/doc/current/setup.html) <br> <br>
[StimulusBundle: Symfony integration with Stimulus](https://symfony.com/bundles/StimulusBundle/current/index.html) <br> <br>
[StackOverflow - Bootstrap cannot find symfony/stimulus-bundle](https://stackoverflow.com/questions/77684268/bootstrap-cannot-find-symfony-stimulus-bundle)

**Comment by MatTheCat** [comment]
The StackOverflow question you provide already had [an accepted answer](https://stackoverflow.com/a/77720816/465233) which seems to make sense. Did you try configuring or removing the AssetMapper component?

**Reply by BAD-WOLF** [reply]
Understood, it seems we're dealing with some challenges here. Let's recap:

**Initial Issue:**
Am I correctly understanding that I'm unable to use AssetMapper? I haven't removed it yet in version 7.0, but in the previous version without AssetMapper, everything worked fine. However, now I've noticed the absence of stimulus-bundle, which in version 6.4 was replaced by stimulus-bridge. The actual error is related to the absence of stimulus-bundle:

> "Module build failed: Module not found: @symfony/stimulus-bundle"

This signals that the problem lies in not locating the package. Additionally, I observed that --webapp includes stimulus-bundle in assets/bootstrap.js, rendering reverting to Symfony skeleton version 6.4 ineffective.

**Attempted Alternative:**
I tried an alternative solution by modifying the import-from statement in bootstrap.js, changing from stimulus-bundle to stimulus-bridge:

```diff
-import { startStimulusApp } from '@symfony/stimulus-bundle';
+import { startStimulusApp } from '@symfony/stimulus-bridge';
 
 const app = startStimulusApp();
 // register any custom, 3rd party controllers here
```

However, this change resulted in another error:

![Error](https://github.com/symfony/symfony/assets/88906998/bf6c245a-b32c-4cbc-a8c0-fef9522cbcef)

We're facing an additional hurdle. Any suggestions on how to proceed from here?

**Reply by MatTheCat** [reply]
> Am I correctly understanding that I'm unable to use AssetMapper?

It’s not that you cannot use it, but it’s supposed to avoid using a bundler so I guess it’s expected it does not work with webpack. In a new symfony webapp installation you don’t need a build step: `@symfony/stimulus-bundle` is resolved through an `importmap.php` file to a file located in your vendors. Webpack won’t know this mapping so it will fail.

Regarding `stimulus-bridge` it seems the error tells you what to do?

**Reply by BAD-WOLF** [reply]
## VIDEO SHOWING WITHOUT WEBPACK
[![Vídeo](https://github.com/symfony/symfony/assets/88906998/cce0fb20-9e86-4ab6-9826-688f5d1d360e)](https://files.catbox.moe/85bv8g.mp4)

**Reply by MatTheCat** [reply]
Err you ask for a `home` controller while only having a `hello` controller.

Anyways seeing that video it seems you managed to use the AssetMapper so I guess your problem is solved?

**Reply by BAD-WOLF** [reply, marked as answer]
I understand, besides the error you told me
> "Err you ask for a home controller while only having a hello controller."

also find out something else I was doing wrong

``` twig
{% block body %}
    {{ stimulus_controller('hello') }}
{% endblock %}
```

In this case, the result obtained is in fact the string 'data-controller'hello'. Since the controller was not linked to an HTML element, but rather written in the "body" of the document, the correct way to use it would be:

``` twig
{% block body %}
    <div {{ stimulus_controller('hello') }}>...</div>
{% endblock %}
```

![image](https://github.com/symfony/symfony/assets/88906998/2531c9d7-a0f6-44b5-ad60-f2d5f7297259)

#### Thank you for your attention and help

**Comment by Raghav9888** [comment]
Please help me 
it gives me the error Module not found. in bootstrap.js
how I fix this one. because '@hotwired/stimulus' was already pre-install when we installed the new project
$ yarn watch
yarn run v1.22.19
$ encore dev --watch
Running webpack ...

 ERROR  Failed to compile with 2 errors                                                                                                                                                                10:27:11 am

Module build failed: Module not found:
"./assets/bootstrap.js" contains a reference to the file "@symfony/stimulus-bundle".
This file can not be found, please check it for typos or update it if the file got moved.

"./assets/bootstrap.js" contains a reference to the file "@symfony/stimulus-bridge/lazy-controller-loader".
This file can not be found, please check it for typos or update it if the file got moved.

Entrypoint app/js 15.6 KiB = runtime.js 12.8 KiB app/js.js 2.86 KiB
webpack compiled with 2 errors

**Reply by BAD-WOLF** [reply]
Have you tried downloading the module yet? 
If it is already in importmap.php just do:
`./bin/console importmap:install`

**Reply by alkinbg** [reply]
If you're using WebpackEncore, the bootstrap.js file works in partnership with [@symfony/stimulus-bridge](https://github.com/symfony/stimulus-bridge). With AssetMapper, the bootstrap.js file works directly with this bundle: a @symfony/stimulus-bundle entry is added to your importmap.php file via Flex, which points to a file that is dynamically built to find and load your controllers (see [Configuration](https://symfony.com/bundles/StimulusBundle/current/index.html#configuration)).
[With WebpackEncoreBundle](https://symfony.com/bundles/StimulusBundle/current/index.html#with-webpackencorebundle)
`// webpack.config.js
.enableStimulusBridge('./assets/controllers.json')`

`// assets/bootstrap.js
import { startStimulusApp } from '@symfony/stimulus-bridge';

// Registers Stimulus controllers from controllers.json and in the controllers/ directory
export const app = startStimulusApp(require.context(
    '@symfony/stimulus-bridge/lazy-controller-loader!./controllers',
    true,
    /\.[jt]sx?$/
));`
